### PR TITLE
feat(graphql): more security : with max query depth and max query complexity

### PR DESCRIPTION
**Pull request by mauriau**
| Q             | A
| ------------- | ---
| Branch?       | main
| Tickets       | -
| License       | MIT
| Doc PR        | https://github.com/api-platform/docs/pull/2107

On our GraphQL APi, we run [GraphQL COP](https://github.com/dolevf/graphql-cop) and it's detected some security leak.

So I tried to fix some of this like "Alias overloading" and "Field Duplication". 

[Webonyx](http://webonyx.github.io/graphql-php/security/) has this following rules :  QueryComplexity and QueryDepth. So I implemented this on Api Platform to be configurable in api_platform.yml. And setted by default to 100

`````
api_platform:
  graphql:
     max_query_depth: 20
     max_query_complexity: 500
`

### CODE REVIEW

**Comment by masseuro** [comment]
Really important for avoid dos attack

**Comment by mauriau** [comment]
🤔 Hmm, the scenario "Introspect the GraphQL schema" from `features/graphql/introspection.feature`  need a max query depth configured to 200. But I dont want to put this value as default value. How can I change this value only for the introspection test ?

**Comment by mauriau** [comment]
> 🤔 Hmm, the scenario "Introspect the GraphQL schema" from `features/graphql/introspection.feature` need a max query depth configured to 200. But I dont want to put this value as default value. How can I change this value only for the introspection test ?

I changed the value in `tests/Fixtures/app/AppKernel.php` to 200 to pass the introspection tests

File: src/Laravel/config/api-platform.php [hunk 1]
```diff
@@ -88,22 +99,22 @@
...
     ],
 
     'url_generation_strategy' => UrlGeneratorInterface::ABS_PATH,
 
     'serializer' => [
         'hydra_prefix' => false,
         // 'datetime_format' => \DateTimeInterface::RFC3339
-    ]
+    ],
 ];
```
Comment (soyuka) [inline src/Laravel/config/api-platform.php:120]: I see this file modified but it looks like cs fixes that I should probably do in another PR. Would you be able to add the configuration at https://github.com/api-platform/core/blob/main/src/Laravel/ApiPlatformProvider.php#L1295 as well?

File: src/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 2]
```diff
@@ -308,7 +311,7 @@ private function addSwaggerSection(ArrayNodeDefinition $rootNode): void
                             ->end()
                             ->validate()
                                 ->ifTrue(static fn ($v): bool => $v !== array_intersect($v, $supportedVersions))
-                                ->thenInvalid(sprintf('Only the versions %s are supported. Got %s.', implode(' and ', $supportedVersions), '%s'))
+                                ->thenInvalid(\sprintf('Only the versions %s are supported. Got %s.', implode(' and ', $supportedVersions), '%s'))
```
Comment (soyuka) [inline src/Symfony/Bundle/DependencyInjection/Configuration.php:314]:
```suggestion
                                ->thenInvalid(sprintf('Only the versions %s are supported. Got %s.', implode(' and ', $supportedVersions), '%s'))
```

File: src/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 3]
```diff
@@ -83,7 +82,7 @@ public function getConfigTreeBuilder(): TreeBuilder
                     ->defaultValue('0.0.0')
                 ->end()
                 ->booleanNode('show_webby')->defaultTrue()->info('If true, show Webby on the documentation page')->end()
-                ->booleanNode('use_symfony_listeners')->defaultFalse()->info(sprintf('Uses Symfony event listeners instead of the %s.', MainController::class))->end()
+                ->booleanNode('use_symfony_listeners')->defaultFalse()->info(\sprintf('Uses Symfony event listeners instead of the %s.', MainController::class))->end()
```
Comment (soyuka) [inline src/Symfony/Bundle/DependencyInjection/Configuration.php:85]:
```suggestion
                ->booleanNode('use_symfony_listeners')->defaultFalse()->info(sprintf('Uses Symfony event listeners instead of the %s.', MainController::class))->end()
```

File: src/Symfony/Bundle/DependencyInjection/Configuration.php [hunk 4]
```diff
@@ -524,7 +527,7 @@ private function addExceptionToStatusSection(ArrayNodeDefinition $rootNode): voi
                         ->then(static function (array $exceptionToStatus): array {
                             foreach ($exceptionToStatus as $httpStatusCode) {
                                 if ($httpStatusCode < 100 || $httpStatusCode >= 600) {
-                                    throw new InvalidConfigurationException(sprintf('The HTTP status code "%s" is not valid.', $httpStatusCode));
+                                    throw new InvalidConfigurationException(\sprintf('The HTTP status code "%s" is not valid.', $httpStatusCode));
```
Comment (soyuka) [inline src/Symfony/Bundle/DependencyInjection/Configuration.php:530]:
```suggestion
                                    throw new InvalidConfigurationException(sprintf('The HTTP status code "%s" is not valid.', $httpStatusCode));
```

**Review by soyuka — CHANGES_REQUESTED** [review]
Nice one, just a few comments, thanks for this addition!

File: src/Laravel/config/api-platform.php (hunk 1 above)
Comment (mauriau) [reply src/Laravel/config/api-platform.php]: Yes, cs fixer did it :)

**Comment by mauriau** [comment]
@soyuka I have some failed on behat's tests (features/hydra/docs.feature:10), but I don't know why

**Comment by soyuka** [comment]
Thanks!
